### DISCUSSION TITLE
How to use custom clang-tidy plugins?

### DISCUSSION THREAD

**Discussion by scarf005** [clangd/clangd#1721] [Q&A]
read both clang-tidy and clangd docs and `--help` output but there doesn't seem be a way to either pass clang-tidy flags or set custom plugin path.

for example, it's possible to invoke clang-tidy with custom plugin like this:

```
/bin/ls -1 src/*.cpp | rargs clang-tidy --load=build/tools/clang-tidy-plugin/libCataAnalyzerPlugin.so --checks='-*,cata-static-declarations' {} --fix
```

however afaik there's no way to indicate clangd that such plugin exists.

**Comment by HighCommander4** [comment]
This is not currently supported; see https://github.com/clangd/clangd/issues/1458 for previous discussion of this.
